src/microsoft.csharp/src/microsoft/csharp/runtimebinder/semantics/types/typeparametertype.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CSharp.RuntimeBinder.Semantics
+{
+    /////////////////////////////////////////////////////////////////////////////////
+
+    internal sealed class TypeParameterType : CType
```

Comment:
If I understand this PR correctly, why keeping `TypeParameterType` and `TypeParameterSymbol` classes? shouldn't they removed with their uses?

Reply:
They're still used in defining types and methods. We can still have a generic method on a dynamic object called, or a dynamic object passed to a generic method, so it needs to start with a TypeParameterType in the definition and then works out the type that parameter needs to be in the call. It also creates generic types open-generic first (changing from that approach is perhaps one possible way to stop the problem that circular type definitions, but I'm not clear on if that's a good idea or not yet).

What we can't have, unless I'm missing something, is casts between type parameters as one can if compiling `(T)value` within a generic method, calling a generic with certain constraints from another generic, etc. which a lot of what this removes relates to. (With a little bit of tidying up of code near where removals had happened).